completion/available/artisan.completion.bash

```diff
@@ -0,0 +1,27 @@
+# shellcheck shell=bash
+cite "about-completion"
+about-completion "Laravel artisan completion"
+
+# Completion function for Laravel artisan
+_artisan_completion() {
+	local cur commands
+	COMPREPLY=()
+	cur="${COMP_WORDS[COMP_CWORD]}"
+
+	# Only provide completions if artisan file exists in current directory or parent directories
+	if [[ ! -f "artisan" ]]; then
+		return 0
+	fi
+
+	# Get list of available artisan commands
+	# Use command prefix to bypass user aliases
+	commands=$(command php artisan --raw --no-ansi list 2> /dev/null | command sed "s/[[:space:]].*//g")
```

Comment:
> `sed "s/[[:space:]].*//g"`

Why is the flag `g` supplied? Once `[[:space:]]` matches, `.*` matches the suffix till the end, so there will be no second or later matches.

Reply:
close conversation